### ISSUE TITLE
master with mdbx does not compile

### ISSUE THREAD

**Issue by myfreeweb** [Kerollmops/heed#45]
```
error[E0308]: mismatched types
   --> /home/greg/.local/share/cargo/git/checkouts/heed-e741d25df84a0eb8/755ef1e/heed/src/db/polymorph.rs:219:17
    |
219 |                 txn.txn,
    |                 ^^^^^^^ expected *-ptr, found struct `txn::RoTxn`
    |
    = note: expected raw pointer `*mut mdbx_sys::MDBX_txn`
                    found struct `txn::RoTxn<T>`
```

**Comment by Kerollmops** [comment]
Thank you, this CI problem revealed that there were a segfault on MDBX and this segfault was due to a problem in the abortion part, I wasn't nullifying the transaction pointer and this resulted in a double abortion issue. This is fixed, on master at least.

**Comment by Kerollmops** [comment]
This is fixed and released in v0.8.0.
